src/main/java/com/hotels/bdp/circustrain/bigquery/extraction/container/DeleteTableAction.java

```diff
@@ -25,18 +25,22 @@
   private static final Logger log = LoggerFactory.getLogger(DeleteTableAction.class);
 
   private final Table table;
+  private boolean deleted = false;
 
   public DeleteTableAction(Table table) {
     this.table = table;
   }
 
   @Override
   public void run() {
-    try {
-      table.delete();
-      log.debug("Deleted table '{}'", table.getTableId());
-    } catch (BigQueryException e) {
-      log.error("Could not delete BigQuery table '{}'", table.getTableId(), e);
+    if (!deleted) {
+      try {
+        table.delete();
```

Comment:
Looks like this method returns true if it succeeds or false if the table wasn't found, so you could set `deleted` here, instead of on another line. Then the log could say if table was indeed deleted, or if it was not found.

Reply:
Good spot!